### PR TITLE
Add blog post about what's new in 0.14.0

### PR DESCRIPTION
**Pull request by scholzj**
This blog post gives an introduction to all the new stuff in 0.14.0 release.

### CODE REVIEW

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+One of the reasons might be that you want to have all users managed inside a single system.
+That gives you much better overview of which users exist and what can they do. 
+You can see it all in one place.
+Most systems are also not based only around Apache Kafka.
+Often single applications needs to use Kafka as well as for example some HTTP or other APIs.
+And ideally it should be able to use a single identity for everything.
+
+Kafka already supports authentication using SASL OAUTHBEARER mechanism. 
+This mechanism allows clients to OAuth tokens obtained from and OAuth 2.0 authorization server and use them to authenticate against the Kafka broker.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
This mechanism allows clients to obtain OAuth tokens from an OAuth 2.0 authorization server and use them to authenticate with the Kafka broker.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+And in the 0.14.0 release we are adding support for it also in Strimzi.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
Strimzi 0.14.0 adds support for this mechanism.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+To enable the support, we developed our own [callback library](https://github.com/strimzi/strimzi-kafka-oauth) responsible for obtaining and validating the OAuth tokens.
+This library is not used only by the Kafka brokers.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
This library is not only used by the Kafka brokers.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+It is also used in all the other components which Strimzi supports and which connect to the Kafka brokers as clients such as Connect, Mirror Maker or our HTTP Bridge.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
It is also used in all the other components which Strimzi supports and which connect to the Kafka brokers as clients, such as Connect, Mirror Maker or our HTTP Bridge.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+This library is also available in [Maven repositories](https://search.maven.org/artifact/io.strimzi/oauth/0.1.0/pom), so that you can use it in your own clients using the Kafka Producer, Consumer and Streams APIs.
+
+In 0.14.0 we have just started with OAuth support.
+We did extensive testing with selected OAuth 2.0 authorization servers such as [Keycloak](https://www.keycloak.org/).
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]: We should mention hydra too.

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+But we hope that you as our community will provide us with additional feedback about this implementation - be it missing features or just letting us know how it works with your OAuth 2.0 server.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
But we hope that you, our community, will provide us with additional feedback about this implementation - be it missing features or just letting us know how it works with your OAuth 2.0 server.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+For the next releases we also plan to add support for authorization using OAuth.
+Until then, you can use the User Operator to manage the ACL rights of the OAuth user.
+
+In the coming days, we will publish a dedicated blogpost explaining how to setup and use OAuth authentication in Strimzi.
+
+## Kafka Exporter
+
+Kafka natively supports JMX metrics.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
Kafka natively exposes metrics via JMX.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+In Strimzi we use the [Prometheus JMX Exporter](https://github.com/prometheus/jmx_exporter) project to take the built in JMX metrics and export them as Prometheus metrics, which can be easily used in many Kubernetes environments.
+However, there are some useful metrics which are missing Kafka.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
However, there are some useful metrics which Kafka lacks.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+For example metrics around consumer lag, offsets and more details metrics about topics and partitions.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
For example metrics around consumer lag, offsets and more detailed metrics about topics and partitions.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+These metrics are very important for monitoring not just your brokers but also your clients.
+
+[Kafka Exporter](https://github.com/danielqsj/kafka_exporter) is a well know project from Daniel Qian.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
[Kafka Exporter](https://github.com/danielqsj/kafka_exporter) is a well-known project from Daniel Qian.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+It connects to Kafka as a client and collects different information about topics, partitions and consumer groups.
+It then exposes these information as Prometheus metric endpoint.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
It then exposes this information as a Prometheus metric endpoint.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+You can scrape these metrics together with the other Kafka metrics into your Prometheus instance and from there you can use them in your alerts or dashboards.
+To make this easier, we provide sample [Prometheus Alerts](https://github.com/strimzi/strimzi-kafka-operator/blob/master/metrics/examples/prometheus/install/prometheus-rules.yaml) and [Grafana](https://github.com/strimzi/strimzi-kafka-operator/blob/master/metrics/examples/grafana/strimzi-kafka-exporter.json) dashboard using the new metrics.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
To make this easier, we provide sample [Prometheus Alerts](https://github.com/strimzi/strimzi-kafka-operator/blob/master/metrics/examples/prometheus/install/prometheus-rules.yaml) and a [Grafana](https://github.com/strimzi/strimzi-kafka-operator/blob/master/metrics/examples/grafana/strimzi-kafka-exporter.json) dashboard using the new metrics.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+In Strimzi 0.14.0 you can now easily deploy Kafka Exporter as a part of your Kafka cluster.
+All you need is to specify it in your Kafka custom resource.
+In the coming days, we will publish a dedicated blogpost explaining why is Kafka Exporter important, what metrics does it provide and how to deploy it.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
In the coming days, we will publish a dedicated blogpost explaining why Kafka Exporter is important, what metrics  it provides and how to deploy it.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+## Tracing
+
+Observability is one of the big topics of the last years.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
Observability has been an a big topic in recent years.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+It came with the rise of micro-service based architectures and was heavily enabled by tools such as the Envoy proxy and all the different projects building on top of it.
+Tracing is one of the important parts of observability.
+
+In 0.14.0 we add support for tracing into several components supported by Strimzi.
+As a basis for tracing in Strimzi, we use the two well known CNCF projects - [OpenTracing](https://opentracing.io/docs/overview/what-is-tracing/) and [Jaeger](https://www.jaegertracing.io/).
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
As a basis for tracing in Strimzi, we use the two well-known CNCF projects - [OpenTracing](https://opentracing.io/docs/overview/what-is-tracing/) and [Jaeger](https://www.jaegertracing.io/).
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+It is supported in the Kafka components based on Kafka clients - such as Mirror Maker and Connect.
+The Strimzi Bridge doesn't yet support tracing in 0.14.0, but we plan to add it in one of the next versions.
+The built in support for tracing makes it easy for you to enable it when deploying your applications.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
The built-in support for tracing makes it easy for you to enable it when deploying your applications.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+You can use tracing in Kafka clients, but having a support for it in the components such as Connect or Mirror Maker will ad more trace points to your application landscape and will help you monitor the data flows and performance.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
You can use tracing in Kafka clients, but having support for it in the components such as Connect or Mirror Maker will add more trace points to your application landscape and will help you monitor the data flows and performance.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+
+## HTTP Bridge
+
+There are no major changes in the HTTP Bridge in 0.14.0 release.
+But we continued to work on the Bridge fixed many bugs, improved the test coverage and optimized the code for better performance.
+You can refer to the Strimzi Bridge release [change log](https://github.com/strimzi/strimzi-kafka-bridge/releases/tag/0.14.0) for more details.
+
+## Environment variables
+
+Using an operator such as Strimzi makes it sometimes hard to customize your deployment and make it run exactly as you want.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
Using an operator such as Strimzi can make it harder to fine-tune your deployment and make it run _exactly_ as you want.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+We are always trying to give our users as much flexibility as possible.
+And that is why the 0.14.0 release adds a possibility to specify environment variables which will be defined inside the containers.
+
+While most users do not care about this feature, there are some use cases where it gets handy.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
While most users do not care about this feature, there are some use cases where it comes in handy.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+For example:
+
+* Enabling some log collection systems such as Sematext which require their own environment variables to be defined inside the containers to enable the log collection.
+* Changing the timezone which is used in the container by defining the `TZ` environment variable.
+Regardless whether you thing it is a good idea to have logs with timestamps from your local timezone or whether you prefer everything in UTC, you have now the choice.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
Regardless whether you think it is a good idea to have logs with timestamps from your local timezone or whether you prefer everything in UTC, you have now the choice.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+
+## Conclusion
+
+This release represents big milestone for Strimzi.
```
Comment (tombentley) [inline _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]:
```suggestion
This release represents another big milestone for Strimzi.
```

File: _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md (hunk 5 above)
Comment (scholzj) [reply _posts/2019-09-30-whats-new-in-strimzi-0.14.0.md]: I do not think we really tested it as working. Unless I missed something on Friday.

**Pull request merged**